hazelcast-jclouds/src/test/java/com/hazelcast/jclouds/JCloudsDiscoveryStrategyTest.java

```diff
@@ -0,0 +1,100 @@
+package com.hazelcast.jclouds;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.nio.Address;
+import com.hazelcast.spi.discovery.DiscoveryNode;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.jclouds.compute.domain.NodeMetadata;
+import org.jclouds.compute.domain.internal.NodeMetadataImpl;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class JCloudsDiscoveryStrategyTest extends HazelcastTestSupport {
+    private static final Set<NodeMetadata> nodes = new HashSet<NodeMetadata>();
```

Comment:
Can't this cause problems? Because these fields are static, so shared between threads. And the tests of this class can be run in parallel, so potentially multiple threads on the same HashSet(s).